### PR TITLE
Add an EVM to the frontend JS context

### PR DESCRIPTION
**Pull request by yijiasu-crypto**

### CODE REVIEW

File: chain/contracts/RoomStateMachine.sol [hunk 1]
```diff
@@ -98,30 +104,52 @@ contract RoomStateMachine {
...
+        judge = BaccaratJudge(judgeAddress);
 
-        if (judgeAddress != address(0)) {
-            judge = BaccaratJudge(judge);
+    }
+
+    // View functions
+
+    function isRoomFull() public view returns (bool) {
+        return players.length == params.playersNum;
```
[the remark this answers is not in the document]
Comment (yijiasu-crypto) [reply chain/contracts/RoomStateMachine.sol:114]: Acknowledge this now

[the remark this answers is not in the document]
Comment (yijiasu-crypto) [reply chain/contracts/RoomStateMachine.sol:114]: Acknowledge this now

File: chain/contracts/RoomStateMachine.sol [hunk 2]
```diff
@@ -206,6 +208,8 @@ contract RoomStateMachine is RoomStateMachineStorage {
         address signer = recoverSigner(messageHash, signature);
         // verify signer
         require(players[playerIndex].playerAddress == signer, "invalid player message");
+
+        // TODO: all successful actions should emit an event so the off-chain EVM can monitor the progress
```
Comment (yihuang) [inline chain/contracts/RoomStateMachine.sol:212]: I just added `StateTransition` and `CardOpened` events in main branch.

**Comment by yijiasu-crypto** [comment]
@yihuang function names are all changed to camelCase in this PR pls be informed

**Comment by yijiasu-crypto** [comment]
@yihuang function names are all changed to camelCase in this PR pls be informed

**Pull request merged**